Services/CO.CDP.Organisation.WebApi.Tests/Api/OrganisationEndpointsTests.cs

```diff
@@ -392,6 +393,117 @@ public async Task GetOrganisationSearch_ReturnsBadRequest_WhenThresholdIsAboveOn
+            {
+                Id = Guid.NewGuid(),
+                Identifiers = new List<Identifier>
+                {
+                    new Identifier { Scheme = "scheme", Id = "123", LegalName = "Test Organisation" }
+                },
+                Name = "Test Organisation",
+                Type = OrganisationType.Organisation,
+                Roles = new List<PartyRole> { PartyRole.Buyer },
+                Addresses = new List<OrganisationAddress>()
+            }
+        };
+
+        _searchByNameOrPponUseCase.Setup(uc => uc.Execute(It.IsAny<OrganisationSearchByPponQuery>()))
+            .ReturnsAsync(searchResults);
+
+        var factory = new TestAuthorizationWebApplicationFactory<Program>(
+            channel, organisationId, organisationPersonScope,
+            services => services.AddScoped(_ => _searchByNameOrPponUseCase.Object),
+            personScope);
+
+        var response = await factory.CreateClient().GetAsync($"/organisation/search-by-name-or-ppon?name=test&limit=10&skip=0");
+
+        response.StatusCode.Should().Be(expectedStatusCode);
+
+        if (expectedStatusCode != Forbidden)
+        {
+            var results = await response.Content.ReadFromJsonAsync<List<OrganisationSearchByPponResult>>();
+            results.Should().BeEquivalentTo(searchResults, options => options.ComparingByMembers<OrganisationSearchByPponResult>());
+        }
+    }
+
+    [Fact]
+    public async Task GetOrganisationSearchByNameOrPpon_Returns404_WhenNoResultsFound()
+    {
+        var organisationId = Guid.NewGuid();
+        var searchResults = new List<OrganisationSearchByPponResult>();
+
+        _searchByNameOrPponUseCase.Setup(uc => uc.Execute(It.IsAny<OrganisationSearchByPponQuery>()))
+            .ReturnsAsync(searchResults);
+
+        var factory = new TestAuthorizationWebApplicationFactory<Program>(
+            Channel.OneLogin, organisationId, OrganisationPersonScope.Admin,
+            services => services.AddScoped(_ => _searchByNameOrPponUseCase.Object));
+
+        var response = await factory.CreateClient().GetAsync($"/organisation/search-by-name-or-ppon?name=nonexistent&limit=10&skip=0");
+
+        response.StatusCode.Should().Be(NotFound);
+    }
+
+    [Theory]
+    [InlineData("test org", 10, 0)]
+    [InlineData("PPON12345", 5, 10)]
+    [InlineData("partial name", 20, 20)]
+    public async Task GetOrganisationSearchByNameOrPpon_UsesCorrectParameters(string searchTerm, int limit, int skip)
+    {
+        var organisationId = Guid.NewGuid();
+        var expectedQuery = new OrganisationSearchByPponQuery(searchTerm, limit, skip);
+
+        var searchResults = new List<OrganisationSearchByPponResult>
+        {
+            new OrganisationSearchByPponResult
+            {
+                Id = Guid.NewGuid(),
+                Identifiers = new List<Identifier>
+                {
+                    new Identifier { Scheme = "scheme", Id = "123", LegalName = "Test Organisation" }
```

Comment:
I think test data should be as close to real data as possible. At least in the same format